javalib/src/main/scala/java/lang/Integer.scala

```diff
@@ -1,45 +1,49 @@
 package java.lang
 
-import scalanative.runtime.{divUInt, remUInt}
+import scalanative.runtime.{divUInt, remUInt, intToULong, Intrinsics}
 
 final class Integer(override val intValue: scala.Int)
     extends Number
     with Comparable[Integer] {
-  def this(s: String) = this(Integer.parseInt(s))
+  @inline def this(s: String) =
+    this(Integer.parseInt(s))
 
-  @inline override def byteValue(): scala.Byte   = intValue.toByte
-  @inline override def shortValue(): scala.Short = intValue.toShort
-  @inline def longValue(): scala.Long            = intValue.toLong
-  @inline def floatValue(): scala.Float          = intValue.toFloat
-  @inline def doubleValue(): scala.Double        = intValue.toDouble
+  @inline override def byteValue(): scala.Byte =
+    intValue.toByte
 
-  @inline override def equals(that: Any): scala.Boolean = that match {
-    case that: Integer => byteValue == that.intValue
-    case _             => false
-  }
+  @inline override def shortValue(): scala.Short =
+    intValue.toShort
+
+  @inline def longValue(): scala.Long =
+    intValue.toLong
+
+  @inline def floatValue(): scala.Float =
+    intValue.toFloat
 
-  @inline override def hashCode(): Int =
+  @inline def doubleValue(): scala.Double =
+    intValue.toDouble
+
+  @inline override def equals(that: Any): scala.Boolean =
+    that match {
+      case that: Integer =>
+        byteValue == that.intValue
+      case _ =>
+        false
+    }
+
+  @inline override def hashCode(): scala.Int =
     Integer.hashCode(intValue)
 
-  @inline override def compareTo(that: Integer): Int =
+  @inline override def compareTo(that: Integer): scala.Int =
     Integer.compare(intValue, that.intValue)
 
   @inline override def toString(): String =
     Integer.toString(intValue)
 
-  /* Methods of java.lang.Byte and java.lang.Short.
-   * When calling a method of j.l.Byte or j.l.Short on a primitive value,
-   * it appears to be called directly on the primitive value, which has type
-   * IntType. Call resolution, by the analyzer and the optimizer, will then
-   * look for the method in the class j.l.Integer instead of j.l.Byte or
-   * j.l.Short. This is why we add here the methods of these two classes that
-   * are not already in j.l.Integer.
-   */
-
-  @inline def compareTo(that: Byte): Int =
+  @inline def compareTo(that: Byte): scala.Int =
     Integer.compare(intValue, that.intValue)
 
-  @inline def compareTo(that: Short): Int =
+  @inline def compareTo(that: Short): scala.Int =
     Integer.compare(intValue, that.intValue)
```

Comment:
Remove those two methods. As the comment indicate, they're an artifact of Scala.js not boxing primitives. SN should not have them.